client/client_test.go

```diff
@@ -2944,6 +2945,39 @@ func testCacheMountNoCache(t *testing.T, sb integration.Sandbox) {
 	require.NoError(t, err)
 }
 
+func testCopyFromScratch(t *testing.T, sb integration.Sandbox) {
+	requiresLinux(t)
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	st := llb.Scratch().File(llb.Copy(llb.Scratch(), "/", "/"))
+	def, err := st.Marshal(sb.Context())
+	require.NoError(t, err)
+
+	_, err = c.Solve(sb.Context(), def, SolveOpt{}, nil)
+	require.NoError(t, err)
+
+	st = llb.Scratch().File(llb.Copy(llb.Scratch(), "/foo", "/"))
+	def, err = st.Marshal(sb.Context())
+	require.NoError(t, err)
+
+	_, err = c.Solve(sb.Context(), def, SolveOpt{}, nil)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "/foo: no such file or directory")
```

Comment:
regarding errors, if I change `llb.Scratch()` to `llb.Image("tonistiigi/test:nolayers")` I get "failed to solve: rpc error: code = Unknown desc = failed to compute cache key: cannot checksum empty reference" here.